cluster/src/main/java/com/hartwig/pipeline/transfer/sbp/SbpFileTransfer.java

```diff
@@ -60,33 +62,56 @@ public void publish(SomaticRunMetadata metadata, SbpRun sbpRun, String sbpBucket
             }
         }
 
-        String manifest = writeManifest(allFiles, metadata.runName());
-        Blob manifestBlob = sourceBucket.get(manifest);
-        if (manifestBlob == null) {
-            throw new RuntimeException("Failed to upload locally-generated manifest");
-        } else {
-            allFiles.add(createPair(manifestBlob, sbpBucket));
-        }
-
+        writeManifest(allFiles, sbpBucket, metadata.runName(), sbpRun);
         for (SourceDestPair pair : allFiles) {
             LOGGER.debug("Copying {}", pair);
-            cloudCopy.copy(pair.source.toUrl(), pair.dest.toUrl());
-            sbpS3.setAclsOn((pair.dest.bucket()), pair.dest.path());
-            SbpFileMetadata metaData = SbpFileMetadata.builder()
-                    .directory(extractDirectoryNameForSbp(pair.dest.path()))
-                    .run_id(Integer.parseInt(sbpRun.id()))
-                    .filename(new File(pair.dest.path()).getName())
-                    .filesize(pair.source.size())
-                    .hash(convertMd5ToSbpFormat(pair.source.md5()))
-                    .build();
-            sbpApi.postFile(metaData);
+            doRemoteWork(pair.source.toUrl(), Integer.parseInt(sbpRun.id()), pair.source.size(), pair.source.md5(), pair.dest);
         }
 
         if (arguments.cleanup()) {
             sourceObjects.forEach(Blob::delete);
         }
     }
 
+    private void writeManifest(List<SourceDestPair> allFiles, String sbpBucket, String directory, SbpRun sbpRun) {
```

Comment:
It looks like the manifest will only get written to S3 and not to GCP? Why the distinction? 

Also good to keep in mind that there will be a time when we only transfer to GCP, but still need to update the SBP API with files/md5s/etc for data sharing. So I guess quite a bit of this will change when that happens.